core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1048,7 +1065,7 @@ private[spark] class BlockManager(
           try {
             replicate(blockId, bytesToReplicate, level, remoteClassTag)
           } finally {
-            bytesToReplicate.dispose()
+            bytesToReplicate.unmap()
```

Comment:
one new thought: ideally we only wanna dispose bytes returned by `DiskStore`. `ChunkedByteBuffer` has a boolean flag, `disposed`. If this flag is true, `ChunkedByteBuffer.dispose` will become no-op.

When `MemoryStore` returns a `ChunkedByteBuffer`, how about we setting the `disposed` flag to true? Then we won't dispose it later.`